hnn_core/params_schema.json

```diff
@@ -0,0 +1,1962 @@
+                1.0
+            ]
+        },
+        "gbar_evprox_2_L2Pyr_ampa": {
+            "$id": "#/properties/gbar_evprox_2_L2Pyr_ampa",
+            "type": "number",
+            "title": "The Gbar_evprox_2_l2pyr_ampa Schema",
+            "description": "An explanation about the purpose of this instance.",
+            "default": 0,
+            "examples": [
+                1.43884
+            ]
+        },
+        "gbar_evprox_2_L2Pyr_nmda": {
+            "$id": "#/properties/gbar_evprox_2_L2Pyr_nmda",
+            "type": "number",
+            "title": "The Gbar_evprox_2_l2pyr_nmda Schema",
+            "description": "An explanation about the purpose of this instance.",
+            "default": 0,
+            "examples": [
+                0.0
+            ]
+        },
+        "gbar_evprox_2_L2Basket_ampa": {
+            "$id": "#/properties/gbar_evprox_2_L2Basket_ampa",
+            "type": "number",
+            "title": "The Gbar_evprox_2_l2basket_ampa Schema",
+            "description": "An explanation about the purpose of this instance.",
+            "default": 0,
+            "examples": [
+                3e-06
+            ]
+        },
+        "gbar_evprox_2_L2Basket_nmda": {
+            "$id": "#/properties/gbar_evprox_2_L2Basket_nmda",
+            "type": "number",
+            "title": "The Gbar_evprox_2_l2basket_nmda Schema",
+            "description": "An explanation about the purpose of this instance.",
+            "default": 0,
+            "examples": [
+                0.0
+            ]
+        },
+        "gbar_evprox_2_L5Pyr_ampa": {
+            "$id": "#/properties/gbar_evprox_2_L5Pyr_ampa",
+            "type": "number",
+            "title": "The Gbar_evprox_2_l5pyr_ampa Schema",
+            "description": "An explanation about the purpose of this instance.",
+            "default": 0,
+            "examples": [
+                0.684013
+            ]
+        },
+        "gbar_evprox_2_L5Pyr_nmda": {
+            "$id": "#/properties/gbar_evprox_2_L5Pyr_nmda",
+            "type": "number",
+            "title": "The Gbar_evprox_2_l5pyr_nmda Schema",
+            "description": "An explanation about the purpose of this instance.",
+            "default": 0,
+            "examples": [
+                0.0
+            ]
+        },
+        "gbar_evprox_2_L5Basket_ampa": {
+            "$id": "#/properties/gbar_evprox_2_L5Basket_ampa",
+            "type": "number",
+            "title": "The Gbar_evprox_2_l5basket_ampa Schema",
+            "description": "An explanation about the purpose of this instance.",
+            "default": 0,
+            "examples": [
+                0.008958
+            ]
+        },
+        "gbar_evprox_2_L5Basket_nmda": {
+            "type": "number",
+            "description": "An explanation about the purpose of this instance.",
+            "default": 0.0
+        },
+        "sync_evinput": {
+            "type": ["boolean", "integer"],
```

Comment:
why both?

Reply:
> why both?

Users should be able to set `params['sync_evinput'] = True` or `params['sync_evinput'] = 1` equivalently since python interprets them the same way. The fastjsonschema, however, does not interpret `0` and `1` (e.g., as written in most param files) as boolean values and must be hard-coded to allow for both types.

Reply:
Not the last time I'll be quoting the Zen of Python:

> There should be one-- and preferably only one --obvious way to do it.
> Although that way may not be obvious at first unless you're Dutch.

I would say it's really non-standard to use 0 and 1 for boolean in Python

Reply:
I agree that it is not best-practice to conflate `1` and `true` in python, however, most users know that `1 == True` returns `True`. This is evident by the fact that our current param files store the `sync_evinput` parameter as either `0` or `1`. It would be odd for us to not control for such a use-case given that it would prevent a plausible error.

Reply:
I mean, if the user provides `0` for `sync_evinput`, we simply throw an error and ask them to fix it. What's the potential downside of this? The less (reasonable) options we provide, the easier it's to maintain the code and the find bugs.

Reply:
A downside would be that the user gets annoyed and stops using the software.

Reply:
These are the reasons non-developers dislike command line interfaces. HNN GUI's users will be users of `hnn-core` now. The GUI has a reasonable way of dealing with this, as you can provide feedback instantly of a non-conforming data type. We can handle this in the GUI if you want to enforce this for the param object.

Reply:
In the GUI, you would create the input boxes so that they do not accept any kind of input. Same as you have in web forms that don't accept strings for age etc. It's possible to do, no?

Reply:
By the way, this is a minor point. Let's not [bikeshed](https://en.wikipedia.org/wiki/Law_of_triviality) over it. It's just that I have not seen the use of 0 or 1 for boolean in most Python libraries. And in general I have seen a lot of non-standard ways of doing things in HNN, I'd just like to limit the creativity as much as possible :)